src/ImageReconstruction.jl

```diff
@@ -36,20 +26,20 @@ function radon(image::AbstractMatrix, θ::AbstractRange, t::AbstractRange)
 
             (a < 1 || a > length(t)) && continue
             α = abs(t′ - t[a])
-            P[a, k] += (1 - α) * image[i, j]
+            P[a, k] += (1 - α) * I[i, j]
 
             (a > length(t) + 1) && continue
-            P[a + 1, k] += α * image[i, j]
+            P[a+1, k] += α * I[i, j]
         end
     end
 
     P
 end
 
-function _ramp_spatial(N::Int, τ)
-    h = zeros(N)
+function _ramp_spatial(N::Int, τ, Npad::Int = N)
```

Comment:
The allocation + loop seems like it could be written more concisely as a list comprehension:
```julia
hval(n) = isodd(n) ? -1 / (π * n * τ)^2 : 1 / (4 * τ^2)
h = [hval(i - N2 - 1) for i = 1:N]
```

Reply:
This is a great suggestion. I made a small modification to achieve equivalent results.